projects/graph/graph.py

```diff
@@ -1,48 +1,115 @@
 """
 Simple graph implementation
 """
+
 from util import Stack, Queue  # These may come in handy
 
+
 class Graph:
     """Represent a graph as a dictionary of vertices mapping labels to edges."""
+
     def __init__(self):
         self.vertices = {}
+
     def add_vertex(self, vertex):
         """
         Add a vertex to the graph.
         """
+        self.vertices[vertex] = set()
         pass  # TODO
+
     def add_edge(self, v1, v2):
         """
         Add a directed edge to the graph.
         """
+        if v1 not in self.vertices:
+            self.add_vertex(v1)
+        if v2 not in self.vertices:
+            self.add_vertex(v2)
+
+        self.vertices[v1].add(v2)
+        self.vertices[v2].add(v2)
         pass  # TODO
+
     def bft(self, starting_vertex):
         """
         Print each vertex in breadth-first order
         beginning from starting_vertex.
         """
-        pass  # TODO
+
+        # make a queue
+        queue = Queue()
+        visited = set()
+        # make a visited set
+        # put starting vertex in the queue
+        queue.enqueue(starting_vertex)
+        # while q isn't empty
+        while queue.size():
+                # dequeue
+            node = queue.dequeue()
+            # mark current as visted
+
+            visited.add(node)
+            # for each of the deqeued item's edges
+            for edge in self.vertices[node]:
+                # put them in the queue
+                if edge not in visited:
+                    queue.enqueue(edge)
+
+            pass  # TODO
+
     def dft(self, starting_vertex):
         """
         Print each vertex in depth-first order
         beginning from starting_vertex.
         """
+
+        stack = Stack()
+
+        # make a visited set
+        visited = set()
+        # pyt starting vertext in our stack
+        stack.push(starting_vertex)
+        # while the stack isnt empty
+        while stack.size():
+            node = stack.pop()
+        # pop off the top of the stack, it is our current item
+        # mark it as visted
+            visited.add(node)
+        # for each of our current item's edges
+            for edge in self.vertices[node]:
+                # put them on the stack
+                if edge not in visited:
+                    stack.push(edge)
+            print(visited)
```

Comment:
If you are using the print statement to print the DFT path, you should know that a set will order numbers automatically when they are added to the set. This will break the DFT ordering.

Reply:
I was using the print statement for validation I was getting the visited lol. Another leftover.